### ISSUE TITLE
Handle emojis with multiple skin tones better

### ISSUE THREAD

**Issue by rossmacarthur** [rossmacarthur/emojis#6]
At the moment we currently just add these emojis to the variations but we should return them from `.skin_tones()`.

Most emojis with skin tones have 6 variations but there are some with 26, for example 👩🏿‍❤️‍👨🏼 kinds of emojis.

Questions:
- What should `.skin_tone()` return?
- What should `with_skin_tone()` do in this case?
- Is there a way to make the above ones use multiple skin tones?

**Comment by fruitsbat** [comment]
any news on this?
it'd be great if those would be added to the `SkinTone` enum

**Comment by rossmacarthur** [comment]
@zoe-bat I think it just requires a sensible design for the API, it wouldn't be too hard to actually implement. What is the use case? Do you just want to list all the skin tone variations for the particular emoji?

Are you suggesting we just add them all to the `SkinTone` enum like this?

```rust
enum SkinTone {
    Default,
    Light,
    MediumLight,
    Medium,
    MediumDark,
    Dark,
    LightAndMediumLight,
    LightAndMedium,
    LightAndMediumDark,
    LightAndDark,
    MediumLightAndLight,
    MediumLightAndMedium,
    MediumLightAndMediumDark,
    MediumLightAndDark,
    MediumAndLight,
    MediumAndMediumLight,
    MediumAndMediumDark,
    MediumAndDark,
    MediumDarkAndLight,
    MediumDarkAndMediumLight,
    MediumDarkAndMedium,
    MediumDarkAndDark,
    DarkAndLight,
    DarkAndMediumLight,
    DarkAndMedium,
    DarkAndMediumDark,
}

```

The other idea I was toying with was something like the following:

```rust
enum SkinTone {
    Default,
    Light,
    MediumLight,
    Medium,
    MediumDark,
    Dark,
}

enum SkinTones {
    One(SkinTone),
    Two(SkinTone, SkinTone),
}
```

**Comment by fruitsbat** [comment]
@rossmacarthur haha i have to admit that i was thinking of the first way because that way i'd have to change my code the least

but the second one is definitely **a lot** more elegant and would probably also work better in most cases

and yeah that's basically my use case! get a list of all skin tones and compare it against another list of user selected ones

https://github.com/zoe-bat/kaokao/blob/daf3da0a47dc11d4a5c577b06bd50f85f9a7185c/src/moji.rs#L60b

anyway! the idea with the two different enums sounds great, i'd love that

**Comment by rossmacarthur** [comment]
Hmmm I think the main problem with the nested enum is that it allows the enumeration of states that don't actually exist. E.g.

```rust
// will never exist
SkinTones::Two(SkinTone::Medium, SkinTone::Medium)

// will be this instead
SkinTones::One(SkinTone::Medium)
```

So I'm actually leaning to the big enum now

**Comment by fruitsbat** [comment]
oh right,
i guess u could resolve that by telling people not do do that in the docs but that's not really ideal

and another advantage the big enum would have is that it'd not really add any complexity

maybe it doesn't make sense after all to try and overengineer this one 😔

**Comment by fruitsbat** [comment]
oh my god a vector

**Comment by rossmacarthur** [comment]
Released in v0.6.0
